src/init-action-post-helper.ts

```diff
@@ -182,7 +182,12 @@ export async function run(
     );
   }
 
-  if (process.env["CODEQL_ACTION_EXPECT_UPLOAD_FAILED_SARIF"] === "true") {
+  // We do not delete uploaded SARIFs if we're on a fork, as we're missing the
+  // appropriate permissions.
+  if (
+    process.env["CODEQL_ACTION_EXPECT_UPLOAD_FAILED_SARIF"] === "true" &&
+    repositoryNwo.owner !== "github"
```

Comment:
Is this the correct thing to do?

```suggestion
    repositoryNwo.owner === "github"
```

Reply:
You're right.. I think there's some other logical bug here given that the checks passed with this conditional incorrect. Will do some print debugging.